### PR TITLE
Implement service to update users' profile

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR adds API to update users' profile (such data as login and avatar) and unit tests for this logic.

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@AndriiSameliuk, PTAL.

File: src/test/java/io/javaclasses/hookahpalace/service/UserProfileServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        assertEquals(testUserInfo.avatarImage(), updatedUserInfo.avatarImage(),
+                     "Users' avatar was not updated.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideParametersForIncorrectAvatarUpdate")
+    void testIncorrectAvatarUpdate(UserInfo userInfo, String assertionMessage) {
+        Assertions.assertThrows(ValidationException.class, () -> {
+            UserInfo updatedUserInfo = userProfileService.updateAvatar(userInfo);
```
Comment (AndriiSameliuk) [inline src/test/java/io/javaclasses/hookahpalace/service/UserProfileServiceTest.java]: updatedUserInfo unused, please remove.

File: src/test/java/io/javaclasses/hookahpalace/service/UserProfileServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        assertEquals(testUserInfo.id(), updatedUserInfo.id(), "The updated user has the wrong id.");
+
+        assertEquals(newUserInfo.login(), updatedUserInfo.login(), "Users' login was not updated.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideParametersForIncorrectLoginUpdate")
+    void testIncorrectLoginUpdate(UserInfo userInfo) {
+        Assertions.assertThrows(ValidationException.class, () -> {
+            UserInfo updatedUserInfo = userProfileService.updateLogin(userInfo);
```
Comment (AndriiSameliuk) [inline src/test/java/io/javaclasses/hookahpalace/service/UserProfileServiceTest.java]: updatedUserInfo unused, please remove.

File: src/test/java/io/javaclasses/hookahpalace/service/RegistrationTest.java [hunk 3]
```diff
@@ -28,28 +29,28 @@ void testRegistrationServiceWithInvalidData(UserInfo user, String assertMessage)
...
-                                .withAvatarImage(null)
+                                .withAvatarImage(new File("testimage/correctJpg.jpg"))
                                 .build(),
                         "Exception due to incorrect password wasn't thrown."),
                 Arguments.of(new UserInfoBuilder()
                                 .withId("1")
                                 .withLogin("Login1234")
                                 .withPassword("Password123")
-                                .withAvatarImage(null)
+                                .withAvatarImage(new File("testimage/correctJpg.jpg"))
```
Comment (AndriiSameliuk) [inline src/test/java/io/javaclasses/hookahpalace/service/RegistrationTest.java]: This test checks the incorrect avatar image.

**Pull request merged**